### PR TITLE
[CI:DOCS] GHA: Fix dumb error check

### PR DESCRIPTION
**Pull request by cevich**
Previously the reply JSON was examined for the literal presence of the
string 'error'.  This was intended to catch server or query errors and
the like.  However it's not a sound design as valid/legitimate contents
could potentially contain the string.  Fix this by using the `-e` option
to `jq`, with a filter that should always result in a non-empty/null
match.  If this fails or returns null for some reason, then it's safe to
throw a real error code & message.

**Note:** This script is reused by other repositories to check their cirrus-cron job status.  I haven't observed it actually breaking, but it's certainly a glaring weak-spot @edsantiago noticed in a similar check elsewhere.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by cevich** [comment]
@edsantiago low-priority, PTAL when you get a chance.  [Example where this is called from.](https://github.com/containers/buildah/actions/runs/2665619792)

File: .github/actions/check_cirrus_cron/cron_failures.sh [hunk 1]
```diff
@@ -102,6 +97,16 @@ fi
 #     }
 #   }
 # }
+
+# This should never ever return an empty-list, unless there are no cirrus-cron
+# jobs defined for the repository.  In that case, this monitoring script shouldn't
+# be running anyway.
+if ! jq -e '.data.ownerRepository.cronSettings' ./artifacts/reply.json &> /dev/null
```
Comment (edsantiago) [inline .github/actions/check_cirrus_cron/cron_failures.sh:104]: Suggestion:
```bash
foo='.data.ownerRepository.cronSettings'
if ! jq -e $foo ...
    err "GraphQL reply did not contain $foo"
```

**Review by edsantiago — APPROVED** [review]
LGTM. That check always made me nervous, so thanks for fixing it. Does GHA mean Github Actions?

File: .github/actions/check_cirrus_cron/cron_failures.sh (hunk 1 above)
Comment (cevich) [reply .github/actions/check_cirrus_cron/cron_failures.sh:104]: Yeah that line is getting a bit close to needing a scroll-bar, and that error would be more clear.  Okay I'll fix...

**Comment by cevich** [comment]
Force-push: Rebased and implemented Ed's suggestion.  Thanks @edsantiago

**Comment by cevich** [comment]
> Does GHA mean Github Actions?

Ya, I'm really sick of spelling it out every time :laughing:

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Pull request merged**